packages/shared/src/utils/helpers/duration.ts

```diff
@@ -1,17 +1,66 @@
 import { localize } from '@deriv/translations';
 import { toMoment } from '../date';
 
+type TContract = {
+    max_contract_duration: string;
+    min_contract_duration: string;
+    expiry_type: string;
+    start_type: string;
+};
+
+type TMaxMin = {
+    min: number;
+    max: number;
+};
+
+type TUnit = {
+    text: string;
+    value: string;
+};
+
+type TDurations = {
+    // min_max: {
+    //     spot?: Partial<Record<'tick' | 'intraday' | 'daily', TMaxMin>>;
+    //     forward?: Record<'intraday', TMaxMin>;
+    // };
+    // units_display: Partial<Record<'spot' | 'forward', TUnit[]>>;
+
+    min_max: {
+        // spot: {
+        //     tick: TMaxMin;
+        //     intraday: TMaxMin;
+        //     daily: TMaxMin;
+        // };
+        // forward: {
+        //     intraday: TMaxMin;
+        // };
+    };
+    units_display: {
+        // spot: TUnit[];
+        // forward: TUnit[];
+    };
+};
+
 const getDurationMaps = () => ({
-    t: { display: localize('Ticks'), order: 1 },
+    t: { display: localize('Ticks'), order: 1, to_second: 0 },
     s: { display: localize('Seconds'), order: 2, to_second: 1 },
     m: { display: localize('Minutes'), order: 3, to_second: 60 },
     h: { display: localize('Hours'), order: 4, to_second: 60 * 60 },
     d: { display: localize('Days'), order: 5, to_second: 60 * 60 * 24 },
 });
 
-export const buildDurationConfig = (contract, durations = { min_max: {}, units_display: {} }) => {
-    durations.min_max[contract.start_type] = durations.min_max[contract.start_type] || {};
-    durations.units_display[contract.start_type] = durations.units_display[contract.start_type] || [];
+export const buildDurationConfig = (
+    contract: TContract,
+    durations: TDurations = { min_max: {}, units_display: {} }
```

Comment:
I defined duration type like this:
`min_max: {
        // spot: {
        //     tick: TMaxMin;
        //     intraday: TMaxMin;
        //     daily: TMaxMin;
        // };
        // forward: {
        //     intraday: TMaxMin;
        // };
    };
    units_display: {
        // spot: TUnit[];
        // forward: TUnit[];
    };`

but in some methods I have issue which I don't know the reason! so I commented the types

Reply:
```suggestion
    durations: TDurations = { min_max: {} as TDurations['min_max'], units_display: {} as TDurations['units_display'] }
```